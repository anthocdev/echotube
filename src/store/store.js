import { applyMiddleware, createStore, compose } from "redux";
import { createLogger } from "redux-logger";
import { default as thunk } from "redux-thunk";
import { createPromise } from "redux-promise-middleware";
import axios from "axios";

import reducer from "../reducers";
//Pulling JWT Token from local storage
const jwttoken = localStorage.getItem("JWT_TOKEN");
//Setting JWT Token as common header for axios
axios.defaults.headers.common["Authorization"] = jwttoken;
//Create store with middleware included
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; //Adding redux dev tool compatibility
const middleware = applyMiddleware(createPromise(), thunk, createLogger()); //Adding thunk and console logging for testing as middleware

export default createStore(
  reducer,
  {
    //Initializing auth state if JWT token is available
    auth: {
      jwtToken: jwttoken,
      isAuth: jwttoken ? true : false
    }
  },
  composeEnhancers(middleware)
);
