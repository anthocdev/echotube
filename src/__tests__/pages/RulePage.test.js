import React from "react";
import { Provider } from "react-redux";
import store from "../../store/store";
import { create, act } from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import RulesPage from "../../modules/Rules";

describe("Rules Page", () => {
  let root;

  it("Renders Without Crashing", async () => {
    act(() => {
      root = create(
        <BrowserRouter>
          <Provider store={store}>
            <RulesPage />
          </Provider>
        </BrowserRouter>
      );
    });
  });

  test("Renders Matches Snapshot", async () => {
    act(() => {
      root = create(
        <BrowserRouter>
          <Provider store={store}>
            <RulesPage />
          </Provider>
        </BrowserRouter>
      );
    });
    expect(root).toMatchSnapshot();
  });
});
