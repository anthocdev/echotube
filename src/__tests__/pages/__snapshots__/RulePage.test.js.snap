// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rules Page Renders Matches Snapshot 1`] = `
<div
  className="rulePage"
>
  <div
    className="ruleContent"
  >
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(196, 98, 235, 0.9)",
            "color": "white",
            "fontWeight": "bold",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Terms of Service
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
    <ul
      className="MuiList-root MuiList-padding"
    >
      <li
        className="MuiListItem-root MuiListItem-gutters MuiListItem-divider"
        disabled={false}
      >
        <div
          className="MuiListItemText-root"
          style={
            Object {
              "color": "white",
              "textAlign": "center",
            }
          }
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            We expect you to follow our Terms Of Service. Button above will lead you to it.
          </span>
        </div>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters MuiListItem-divider"
        disabled={false}
      >
        <div
          className="MuiListItemText-root"
          style={
            Object {
              "color": "white",
              "textAlign": "center",
            }
          }
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Don't use our application in any way that may violate YouTube, Google or Deezer Terms of Service.
          </span>
        </div>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters MuiListItem-divider"
        disabled={false}
      >
        <div
          className="MuiListItemText-root"
          style={
            Object {
              "color": "white",
              "textAlign": "center",
            }
          }
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Don't use 3rd party applications to send API requests, this might end your quota and block future requests.
          </span>
        </div>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters MuiListItem-divider"
        disabled={false}
      >
        <div
          className="MuiListItemText-root"
          style={
            Object {
              "color": "white",
              "textAlign": "center",
            }
          }
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Use of racist or generally offensive language in playlist names is forbidden.
          </span>
        </div>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters MuiListItem-divider"
        disabled={false}
      >
        <div
          className="MuiListItemText-root"
          style={
            Object {
              "color": "white",
              "textAlign": "center",
            }
          }
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Explicit, illegal or offensive image use on playlists is forbidden.
          </span>
        </div>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters MuiListItem-divider"
        disabled={false}
      >
        <div
          className="MuiListItemText-root"
          style={
            Object {
              "color": "white",
              "textAlign": "center",
            }
          }
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Misuse of our contact form may result in termination of your account.
          </span>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
